src/GraphQL/Types/GenericGraphTypeFactory.cs

```diff
@@ -0,0 +1,13 @@
+namespace GraphQL.Types;
+
+/// <summary>
+/// A generic factory to create instances of specific TGraphType implementations from parameterless constructors.
+/// </summary>
+public class GenericGraphTypeFactory<TGraphType> : IGraphTypeFactory<TGraphType> where TGraphType : IGraphType
```

Comment:
IMO, `DefaultGraphTypeFactory` is a better name.

Reply:
Agree.  It also matches `DefaultServiceProvider` and `DefaultNameConverter` and stuff